lib/language_filter.rb

```diff
@@ -265,8 +265,20 @@ def replace(word)
     end
 
     def regex_for(chosen_matchlist)
+      name = chosen_matchlist.join('_')
+      regexp = self.class.take_from_matchlist_hash(name)
+      return regexp if regexp
       list = chosen_matchlist.map { |list_item| "#{beg_regex}#{list_item}#{end_regex}" }
-      %r"#{list.join('|')}"i
+      self.class.update_matchlist_hash(name, %r"#{list.join('|')}"i)
+    end
+
+    def self.take_from_matchlist_hash(name)
+      @matchlist_hash ||= {}
+      @matchlist_hash[name]
```

Comment:
Good question! Right now we got two vocabularies. One for SK, and second for AM. This list of words for regexp is coming from default selected vocabulary + our words. And because of this words are predefined, and not created dynamically, we should'n get infinite amount of them.
Also load testing on local machine prove this.